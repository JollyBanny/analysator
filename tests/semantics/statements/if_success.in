var i : boolean = true;

begin
    if i then write('true');
end.