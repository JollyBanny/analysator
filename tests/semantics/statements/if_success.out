program
└──── program block
      ├──── VAR
      │     └──── var declaration
      │           ├──── i
      │           ├──── boolean
      │           └──── TRUE
      └──── Compound statement
            └──── IF
                  ├──── i
                  └──── Call statement
                        └──── write
                              └──── 'true'

┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  builtins                                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  integer                                                                                         │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  double                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  char                                                                                            │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  string                                                                                          │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Type             │  boolean                                                                                         │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│  global                                                                                                              │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│  Var              │  i                          │  boolean                                                           │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
